setup.py

```diff
@@ -104,7 +104,15 @@ def finalize_options(self):
         build_ext.finalize_options(self)
         # Prevent numpy from thinking it is still in its setup process:
         # https://docs.python.org/2/library/__builtin__.html#module-__builtin__
-        __builtins__.__NUMPY_SETUP__ = False
+        try:
+            __builtins__.__NUMPY_SETUP__ = False
+        except:
+            try:
+                # For python 3
+                import builtins
+                builtins.__NUMPY_SETUP__ = False
+            except:
+                print("Skipping numpy hack; if installation fails, try installing numpy first")
```

Comment:
What exception can possibly get raised here?

Assuming that some exception can, then a explicit error message would be better:

```suggestion
                print("Skipping numpy hack; if gensim installation fails, try installing numpy first")
```

Also, may as well output the reason why the hack failed (what exception was caught) and a trace.